src/child.js

```diff
@@ -0,0 +1,45 @@
+var childProcess = require('child_process');
+var fs = require('fs');
+
+var stdoutFile = process.argv[2];
+var stderrFile = process.argv[3];
+var cmd = process.argv[4];
+var argString = process.argv[5];
+var optString = process.argv[6];
+var pipe = process.argv[7];
+
+var opts = JSON.parse(optString);
+var args = JSON.parse(argString);
+
+var c = childProcess.spawn(cmd, args, opts);
+
+// Wait to exit until all other work has been finished, like closing IO streams
+function delayedExit(code) {
+  setTimeout(function () {
+    process.exit(code); // let streams close before ending
```

Comment:
Should this use [`process.exitCode = code`](https://nodejs.org/api/process.html#process_process_exitcode) instead?

Based on the warning from https://nodejs.org/api/process.html#process_process_exit_code

    It is important to note that calling process.exit() will force the process to exit as quickly as possible even if there are still asynchronous operations pending that have not yet completed fully, including I/O operations to process.stdout and process.stderr.

(Which seems to imply it might not actually let the streams close/flush correctly?)